src/frequenz/sdk/timeseries/_power_streamer.py

```diff
@@ -0,0 +1,197 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Fetch and stream the 3-phase power from a source component.
+
+Each phase of the power is fetched from the source component individually and
+then streamed as a 3-phase sample.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import TYPE_CHECKING
+
+from frequenz.channels import Receiver, Sender
+
+from ..actor import ChannelRegistry
+from ..microgrid import connection_manager
+from ..microgrid.component import Component, ComponentCategory, ComponentMetricId
+from ._base_types import Sample, Sample3Phase
+from ._quantities import Power, Quantity
+
+if TYPE_CHECKING:
+    # Imported here to avoid a circular import.
+    from ..actor import ComponentMetricRequest
+
+_logger = logging.getLogger(__name__)
+
+
+class ActivePowerStreamer:
+    """Fetch and stream the 3-phase active power from a source component.
+
+    Example:
+        ```python
+        from datetime import timedelta
+
+        from frequenz.sdk import microgrid
+        from frequenz.sdk.timeseries import ResamplerConfig
+
+        await microgrid.initialize(
+            "127.0.0.1",
+            50051,
+            ResamplerConfig(resampling_period=timedelta(seconds=1))
+        )
+
+        # Get a receiver for the 3-phase active power.
+        # pylint: disable=protected-access
+        active_power_recv = microgrid._active_power().new_receiver()
+
+        async for active_power_sample in active_power_recv:
+            print(active_power_sample)
+        ```
+    """
```

Comment:
this is an ambiguous/confusing interface because the active power values are actually for an arbitrary meter, but active power is not a common quantity like frequency or voltage, and can be different in different places in the microgrid.  So `microgrid._active_power()` without a component id or a location in the graph, tells us nothing.

If you stream the power factor instead using power measurements from the arbitrary meter, then that would be a common value for the whole microgrid.

Reply:
It should probably also be called `_power_3_phase` if we are going this route.

Reply:
For the current use-case we only need the 3-phase power factor. I thought we can compute the 3-phase Power Factor on the application side given that the 3-phase grid current (formula generator) and 3-phase voltage (resampled voltage) are already available in the SDK. So the idea was to create a temporary method `microgrid._active_power()` ("not exposed to the user") to calculate `PF = Active Power / (Voltage . Current)` in the app and drop it once `Power Factor` is available in the SDK
Anyway I think this was just a bad idea given the implications it might cause.

I can try to (1) calculate the Power Factor in the SDK or (2) create the 3-phase power formula generator. The (1) needs to be dropped once the power factor is available in the SDK and (2) could be helpful for another scenarios

Reply:
The (3) is just to wait until power factor is available and just drop this PR

Reply:
You can't use the grid current for this calculation, unless you're also using the 3-phase grid power.  The current `ActivePowerStreamer` streams the 3-phase power of the `first_descendant_component`, which would be grid power only if the location has a grid-side main meter, which is not everywhere.

Reply:
> (1) calculate the Power Factor in the SDK 

for this you could just find the `first_descendant_component` and use raw data from it to do the calculations

> or (2) create the 3-phase power formula generator

I guess with this, you'd make a 3-phase power formula for grid power?

I think both options are fine as a temporary solution, which ever is easier for you.

Reply:
The grid.current is 3-phase current, see https://github.com/frequenz-floss/frequenz-sdk-python/blob/v1.x.x/src/frequenz/sdk/timeseries/grid.py#L99-L118

> I guess with this, you'd make a 3-phase power formula for grid power?

Yes, that's what I have in mind.

I'll give a shot to option (2), that should be the easier one :crossed_fingers:

Reply:
done, implemented option (2)